airflow/www/views.py

```diff
@@ -3527,13 +3527,15 @@ def grid_data(self):
         with create_session() as session:
             query = select(DagRun).where(DagRun.dag_id == dag.dag_id, DagRun.execution_date <= base_date)
 
-        run_type = request.args.get("run_type")
-        if run_type:
-            query = query.where(DagRun.run_type == run_type)
-
-        run_state = request.args.get("run_state")
-        if run_state:
-            query = query.where(DagRun.state == run_state)
+        run_type_raw = request.args.get("run_type")
+        if run_type_raw:
+            run_types = {run_type.strip() for run_type in run_type_raw.split(",")}
+            query = query.where(DagRun.run_type.in_(run_types))
+
+        run_state_raw = request.args.get("run_state")
+        if run_state_raw:
+            run_states = {run_state.strip() for run_state in run_state_raw.split(",")}
+            query = query.where(DagRun.state.in_(run_states))
```

Comment:
To avoid passing these as a big string and parsing this manually in the backend, you can use `request.args.getlist` and pass the query param list in a exploded way. cf query parameters section https://swagger.io/docs/specification/serialization/.

There is no OpenAPI 'standard' for passing list in a query param, and I don't think we have define how to handle this in the webserver yet.

This would just avoid some boilerplate code.

Reply:
I went back and forth with this. The reason I ended up going with CSV here is because it is backwards compatible. For example, Airflow users today might have a URL like this saved:

`my-airflow-server.com/dags/my_dag/grid?run_state=queued`

If we were to change the argument to a list, this URL would be broken. For it to work, it would need to look something like this:

`my-airflow-server.com/dags/my_dag/grid?run_state[]=queued`

By keeping it as a CSV, a single value would look the same as multiple values:

`my-airflow-server.com/dags/my_dag/grid?run_state=queued`
`my-airflow-server.com/dags/my_dag/grid?run_state=queued%2Csuccess%2Crunning`

Reply:
I am not suggesting that we use a query param list with braked notation, exploded would be something like:
`?run_state=queued&run_state=success….`

Then flask handles this nicely with getlist I believe, no need for manual parsing. Cf link above for more information

Reply:
Ok. This will require a bit of refactoring. Give me a couple days.